### DISCUSSION TITLE
Error: I can not change the height when putting it in a class

### DISCUSSION THREAD

**Discussion by moyurun0321** [TomSchimansky/CustomTkinter#1429] [Q&A]
self.radio_button_1 = customtkinter.CTkButton(self, text="Suggestions", command= F1)
        self.radio_button_1.grid(row=1, column=0, width=50, padx=10, pady=10)

This results in the error of

Traceback (most recent call last):
  File "C:\Users\moyur\PycharmProjects\pythonProject\design.py", line 144, in <module>
    app = App()
          ^^^^^
  File "C:\Users\moyur\PycharmProjects\pythonProject\design.py", line 138, in __init__
    self.Tabs = Tabs(self, header_name="Tabs")
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moyur\PycharmProjects\pythonProject\design.py", line 41, in __init__
    self.radio_button_1.grid(row=1, column=0, width=50, padx=10, pady=10)
  File "C:\Users\moyur\AppData\Local\Programs\Python\Python311\Lib\site-packages\customtkinter\windows\widgets\core_widget_classes\ctk_base_class.py", line 321, in grid
    return super().grid(**self._apply_argument_scaling(kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\moyur\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 2549, in grid_configure
    self.tk.call(
_tkinter.TclError: bad option "-width": must be -column, -columnspan, -in, -ipadx, -ipady, -padx, -pady, -row, -rowspan, or -sticky

**Comment by Akascape** [comment]
@Sliseal0324 Why are you adding width in the grid method? 

```python
# design.py, line 41:
self.radio_button_1.grid(row=1, column=0, width=50, padx=10, pady=10)
```

If you want to change parameters of widget then use `self.radio_button_1.configure(width=50, height=50)`

**Comment by moyurun0321** [comment]
oh, I made a mistake. Sorry!!

**Comment by moyurun0321** [comment]
one more question, how should I change the width based on the area of the things below?
![image](https://user-images.githubusercontent.com/107347365/230078783-d7c08831-4138-4c87-a936-3c318dc72d6e.png)

**Comment by moyurun0321** [comment]
Also, I want to hide the previous frame when you click the button, but I do not know how except grid_hide() which does not work
![image](https://user-images.githubusercontent.com/107347365/230080352-35846999-13c2-41b5-92dc-aeb0f0f3b569.png)

**Comment by Akascape** [comment]
@Sliseal0324 

> one more question, how should I change the width based on the area of the things below? 

Use the grid method properly, like adding `sticky='we'` can help you fill the empty space.
You can also use columnconfigure and rowconfigure to manually adjust the size of grid. 

> Also, I want to hide the previous frame when you click the button, but I do not know how except grid_hide() which does not work

You should use `.grid_forget()`.

All these questions are available if you google the same topic for tkinter. Customtkinter works almost the same like tkinter.

**Comment by moyurun0321** [comment]
thank you!!

**Comment by moyurun0321** [comment]
Traceback (most recent call last):
  File "C:\Users\moyur\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 1948, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "C:\Users\moyur\AppData\Local\Programs\Python\Python311\Lib\site-packages\customtkinter\windows\widgets\ctk_button.py", line 553, in _clicked
    self._command()
  File "C:\Users\moyur\PycharmProjects\pythonProject\design.py", line 45, in <lambda>
    self.radio_button_1 = customtkinter.CTkButton(self, text="Suggestions", command=lambda:Loading.grid_remove())
                                                                                           ^^^^^^^^^^^^^^^^^^^^^
TypeError: Grid.grid_remove() missing 1 required positional argument: 'self'

I tried to follow the websites on google, but all of them result in this error and if I include self, the button gets deleted obviously

**Reply by Akascape** [reply]
@Sliseal0324 
In your case, i guess self means the widget you are dealing with. Maybe try `.grid_remove(widget_name)`.

**Reply by moyurun0321** [reply]
I tried this, and this came up

 File "C:\Users\moyur\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 2565, in grid_remove
    self.tk.call('grid', 'remove', self._w)
    ^^^^^^^
AttributeError: type object 'Data' has no attribute 'tk'
